manifest-editor/components/modals/NewCanvasModal.tsx

```diff
@@ -0,0 +1,280 @@
+import { useState, useEffect, useContext } from "react";
+
+import { Input, InputLabel } from "../form/Input";
+import { Button, CalltoButton, SecondaryButton } from "../atoms/Button";
+import { CloseIcon } from "../icons/CloseIcon";
+import { ModalBackground } from "../layout/ModalBackground";
+import { ModalContainer } from "../layout/ModalContainer";
+import { FlexContainer, FlexContainerColumn } from "../layout/FlexContainer";
+import { ModalHeader } from "../atoms/ModalHeader";
+import { HorizontalDivider } from "../atoms/HorizontalDivider";
+
+import ShellContext from "../apps/Shell/ShellContext";
+import { analyse } from "../../helpers/analyse";
+import { ErrorBoundary } from "../atoms/ErrorBoundary";
+import { InformationLink } from "../atoms/InformationLink";
+import { useVault } from "react-iiif-vault";
+import { useManifest } from "../../hooks/useManifest";
+
+var uuid = require("uuid");
+
+export const NewCanvasModal: React.FC<{
+  close: any;
+}> = ({ close }) => {
+  const [inputValue, setInputValue] = useState<string | undefined>();
+  const [inputType, setInputType] = useState<string | undefined>();
+  const [label, setLabel] = useState<string | undefined>();
+  const [width, setWidth] = useState<number | undefined>(2000);
+  const [height, setHeight] = useState<number | undefined>(1000);
+  const [duration, setDuration] = useState<number | undefined>();
+  const [imageServiceJSON, setImageServiceJSON] = useState<any>();
+  const [emptyCanvas, setEmptyCanvas] = useState(false);
+  const vault = useVault();
+
+  const shellContext = useContext(ShellContext);
+  const manifest = useManifest();
+
+  const handleChange = async () => {
+    let inputed: any;
+    if (inputValue) {
+      inputed = await analyse(inputValue);
+      setInputType(inputed?.type);
+    }
+    setLabel(inputed?.label);
+    setHeight(inputed?.height);
+    setWidth(inputed?.width);
+    if (
+      inputed &&
+      !(
+        inputed.type === "Manifest" ||
+        inputed.type === "Image" ||
+        inputed.type === "Collection"
+      )
+    ) {
+      setImageServiceJSON(inputed);
+    }
+    if (inputed && inputed.type === "Image" && !emptyCanvas) {
+      const newCanvas = {
+        id: `vault://${uuid.v4()}`,
+        height: inputed?.height,
+        width: inputed?.width,
+        type: "Canvas",
+        annotations: [
+          {
+            id: `vault://${uuid.v4()}`,
+            type: "Annotation",
+            motivation: "Painting",
+            body: {
+              id: inputValue,
+              type: "Image",
+              format: "image/png",
+              height: inputed?.height,
+              width: inputed?.width,
+            },
+          },
+        ],
+      };
+      const previousItems = manifest?.items;
+      // @ts-ignore
+      previousItems?.push(newCanvas);
+      if (manifest) {
+        vault?.modifyEntityField(manifest, "items", previousItems);
```

Comment:
This won't work as expected in many cases. Vault needs resources in a flat and normalised format:
* no missing properties
* no nested items (`{id: '..', type: 'Annotation'}` and then the full annotation separately)

This is what the error on line 79 was probably saying, as it's expecting an `id/type` only, and for the canvas to already be in Vault.